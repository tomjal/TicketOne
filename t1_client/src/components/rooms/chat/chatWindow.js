import React, { Component } from 'react';

import ChatList from './chatList';

export class ChatWindow extends Component {
  constructor() {
    super();
    this.handleKeyUp = this.handleKeyUp.bind(this);
    this.submitSendMessage = this.submitSendMessage.bind(this);
  }
  submitSendMessage() {
    let message = this.chatInput.value;
    if (message && message.length > 0) {
      this.props.sendMessage(message);
      this.chatInput.value = "";
    }
  }
  handleKeyUp(event) {
    if (event.keyCode === 13) return this.submitSendMessage();
  }
  render() {
    const { id, role, messages } = this.props;
    const chatWindowStyle = { marginTop: "10xp", marginBottom: "10px" };
    const inputSpanStyle = { marginLeft: "10px", marginRight: "10px" };
    const inputBoxStyle = { display: "inline-block", width: "270px", fontSize: "1.5rem", marginRight: "20px" };

    return (
      <div style={chatWindowStyle}>
        <ChatList id={id} role={role} messages={messages} />
        <div style={inputSpanStyle}>
          <input style={inputBoxStyle} type="text" className="form-control"
            onKeyUp={this.handleKeyUp} ref={(input) => { this.chatInput = input; }} />
          <button className="btn btn-default" onClick={this.submitSendMessage}>Send message</button>
        </div>
      </div>
    );
  }
}

export default ChatWindow;
