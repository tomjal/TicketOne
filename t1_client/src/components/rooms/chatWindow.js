import React, { Component } from 'react';

import ChatList from './chatList';

class ChatWindow extends Component {
  submitSendMessage() {
    var message = this.chatInput.value ? this.chatInput.value : "default";
    this.props.sendMessage(message);
    this.chatInput.value = "";
  }
  render() {
    const { messages } = this.props;
    return (
      <div>
        <ChatList messages={messages} />
        <input ref={(input) => { this.chatInput = input; }}></input>
        <button onClick={this.submitSendMessage.bind(this)}>Send</button>
      </div>
    );
  }
}

export default ChatWindow;
