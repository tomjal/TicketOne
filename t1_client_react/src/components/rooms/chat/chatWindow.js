import React, { Component } from 'react';
import { css } from 'aphrodite';

import { chatStyles } from './../../../styles/inline/specificStyles';
import { fontSizes } from './../../../styles/inline/genericStyles';

import ChatList from './chatList';

const KEY_CODES = { ENTER: 13 };

export class ChatWindow extends Component {
  constructor() {
    super();
    this.handleKeyUp = this.handleKeyUp.bind(this);
    this.submitSendMessage = this.submitSendMessage.bind(this);
  }
  submitSendMessage() {
    let message = this.chatInput.value;
    if (message && message.length > 0) {
      this.props.sendMessage(message);
      this.chatInput.value = "";
    }
  }
  handleKeyUp(event) {
    if (event.keyCode === KEY_CODES.ENTER) {
      return this.submitSendMessage();
    }
  }
  render() {
    const { id, role, messages } = this.props;

    return (
      <div className={css(chatStyles.window)}>
        <ChatList id={id} role={role} messages={messages} />
        <div className={css(chatStyles.span)}>
          <input type="text" className={`${css(chatStyles.box, fontSizes.bigger)} form-control`}
            onKeyUp={this.handleKeyUp} ref={(input) => { this.chatInput = input; }} />
          <button className="btn btn-default" onClick={this.submitSendMessage}>
            Send message
          </button>
        </div>
      </div>
    );
  }
}

export default ChatWindow;
