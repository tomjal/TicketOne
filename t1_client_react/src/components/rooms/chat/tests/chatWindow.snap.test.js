import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { ChatWindow } from './../chatWindow';

test('ChatWindow snapshot', () => {
  //const wrap = shallow(<ChatWindow />);

  //expect(shallowToJson(wrap)).toMatchSnapshot();

  //TODO Problems with aphrodite
});