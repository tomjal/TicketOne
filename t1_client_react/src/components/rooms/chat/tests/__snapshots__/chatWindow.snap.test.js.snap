// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatWindow snapshot 1`] = `
<div
  className="chatWindow_ji3w17"
>
  <ChatList />
  <div
    className="actionBlock_1o6grbo"
  >
    Conversation closed. Read only mode
  </div>
</div>
`;

exports[`test ChatWindow snapshot 1`] = `
<div
  className="chatWindow_ji3w17">
  <ChatList />
  <div
    className="actionBlock_1o6grbo">
    <input
      className="chatInput_r3suy5-o_O-bigger_tou7b3 form-control"
      onKeyUp={[Function]}
      type="text" />
    <button
      className="btn btn-default"
      onClick={[Function]}>
      Send message
    </button>
  </div>
</div>
`;
